### ISSUE TITLE
[HOLD for payment 2022-11-16] [$1000] iOS - LHN - The green + button moves

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#10905]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the App/iOS and login with any account
2. Click on FAB button
3. Select New Chat or New Group and then close it

## Expected Result:
The FAB button should not move

## Actual Result:
The FAB button moves down

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- iOS

**Version Number:** 1.1.98.0

**Reproducible in staging?:** Yes
 
**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/189195867-e7b6accc-0664-4dec-8f28-9a32df91cddd.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** @thesahindia & Applause - Internal Team

**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1661937997800789

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by thesahindia** [comment]
Hi, I had reported it [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1661937997800789).
It's more noticeable in iPad

https://user-images.githubusercontent.com/83179295/189351776-bf687433-caf6-41f0-9d40-bb45b2f73def.mov

**Comment by kadiealexander** [comment]
Reassigning as I'm ooo for the next 9 days.

**Comment by jboniface** [comment]
[job post](https://www.upwork.com/jobs/~0114bac5a1cd694de8)

**Comment by Santhosh-Sellavel** [comment]
#### This seems like an issue to me -> https://github.com/Expensify/App/issues/10905#issuecomment-1241927509
https://user-images.githubusercontent.com/85645967/189996312-ccb8f0a2-0be9-4b89-8521-94d549f9c400.mov

#### But not this https://github.com/Expensify/App/issues/10905#issue-1366788309
https://user-images.githubusercontent.com/85645967/189996078-0ca684b5-e1b0-43af-8c10-fa60c6b99c6d.mp4

@Expensify/design Can someone share your thoughts

Also, we have a proposal [here](https://github.com/Expensify/App/issues/10905#issuecomment-1244097571) please checkout the after-fix video is that the expectation?

**Comment by shawnborton** [comment]
@Santhosh-Sellavel I agree with you

**Comment by shawnborton** [comment]
And yes, the proposal vid looks good to me.

**Comment by jboniface** [comment]
@ctkochan22  / @shawnborton  should I hire @vladnobenladen?

**Comment by Santhosh-Sellavel** [comment]
@jboniface Not yet

**Comment by Santhosh-Sellavel** [comment]
Oh, @shawnborton Just confused by your comments.

Do you agree that,

1. the issue needs to be addressed only on iPad or Tablets
2. Phone already looks fine
3. And fixed behavior is expected on iPad/tab alone.
Or Do we need this handled for the phone also?

Please clarify, thanks!

**Comment by thesahindia** [comment]
IIRC the fab used to work fine, it used to float correctly. Maybe there was some regression which caused this.
I think the fab should float when keyboard is open because it's easy to access while chatting.

**Comment by Santhosh-Sellavel** [comment]
> IIRC the fab used to work fine, it used to float correctly. Maybe there was some regression which caused this.
I think the fab should float when keyboard is open because it's easy to access while chatting.

I would also expect the same while chatting fab should float!
cc: @shawnborton

**Comment by shawnborton** [comment]
Hmm I kind of think the FAB should just stay stuck to the bottom of the chats list view.

@thesahindia what exactly do you mean by this:

> IIRC the fab used to work fine, it used to float correctly. Maybe there was some regression which caused this.
I think the fab should float when keyboard is open because it's easy to access while chatting.

Did it used to float right above the keyboard?

**Comment by thesahindia** [comment]
Yeah it used to.

**Comment by shawnborton** [comment]
Got it, so then it sounds like this is a regression?

**Comment by Santhosh-Sellavel** [comment]
> Got it, so then it sounds like this is a regression?

@shawnborton 
I'm not sure, check out the below video behavior was correct at the beginning and end. In between, it was messed up

https://user-images.githubusercontent.com/85645967/190205965-fa633d56-9db8-4be4-8e62-ed951b3b2a7b.MP4

**Comment by thesahindia** [comment]
> Got it, so then it sounds like this is a regression?

Yeah It is a regression, here's a video of this working fine earlier (it's from https://github.com/Expensify/App/issues/10354 which was closed because it wasn't a bug) -

https://user-images.githubusercontent.com/83179295/190206841-3c977a99-b87d-4011-80ce-5b55a8b84687.mov

**Comment by Santhosh-Sellavel** [comment]
Solid that's a regression!

**Comment by shawnborton** [comment]
Got it, okay so yeah it seems like the [video here](https://github.com/Expensify/App/issues/10905#issuecomment-1246997790) is correct behavior then.

**Comment by jliexpensify** [comment]
Hi @thesahindia and @Santhosh-Sellavel - I've just been assigned this as Jeremy has now left and I'm confused as to the next steps.

Can one of you provide a suggestion as to the next steps from a management standpoint? Should I look at hiring someone, or do one of you want to take this on as C+? 

Job Post - https://www.upwork.com/ab/applicants/1569366797023916032/job-details

**Comment by Santhosh-Sellavel** [comment]
This is still open to proposals

**Comment by Santhosh-Sellavel** [comment]
@aimane-chnaif 
Your proposal is redundant as https://github.com/Expensify/App/issues/10905#issuecomment-1244097571

**Comment by Santhosh-Sellavel** [comment]
We are not moving forward as it does not meet our requirements!

**Comment by shawnborton** [comment]
Not overdue

**Comment by ctkochan22** [comment]
Not overdue

**Comment by jliexpensify** [comment]
Going to double this one, since we don't have any accepted proposals

**Comment by shawnborton** [comment]
Not overdue

**Comment by Santhosh-Sellavel** [comment]
@jliexpensify Can we bump this to weekly while we wait, also we can double the price for this one!

**Comment by jliexpensify** [comment]
@Santhosh-Sellavel I made it Weekly, but I only just doubled 3 days ago...so let's wait until next week?

**Comment by Santhosh-Sellavel** [comment]
Ah Sorry, I missed your comment 3 days ago. @jliexpensify

**Comment by jliexpensify** [comment]
Doubled - https://expensify.slack.com/archives/C01GTK53T8Q/p1664851140216699

**Comment by jayeshmangwani** [comment]
### Proposal

we can remove this file https://github.com/Expensify/App/blob/main/src/components/FAB/index.ios.js,
as this file was added for this issue https://github.com/Expensify/App/issues/2151 and code was added in this [PR] for iPad FAB button issue. (https://github.com/Expensify/App/pull/2305/files) , but now we are using `ScreenWrapper` and under this file we are using `KeyboardAvoidingView` so no need to add `KeyboardAvoidingView` for to FAB as parent screen `SidebarScreen` is wrapped in `ScreenWrapper` 

video that shows that after removing file, its working fine in iPhone 13 and iPad pro

https://drive.google.com/file/d/12HCy-tr3GH4Qn-oWR_cZLi2UEcJlnJ9Q/view?usp=sharing

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel can you please clarify in detail what is not met?
> You mean you don't like both Solution 1 and Solution 2?

Can you add a recording of behavior after applying the solutions?

**Comment by jayeshmangwani** [comment]
> Can you elaborate on this
> 
> > but now we are using ScreenWrapper and under this file, we are using KeyboardAvoidingView so no need to add KeyboardAvoidingView for to FAB as parent screen SidebarScreen is wrapped in ScreenWrapper

@Santhosh-Sellavel Yes sure, 

1. we have added KeyboardAvoidingView for FAB in ios specific for iPad in this [PR](https://github.com/Expensify/App/pull/2305/files) and this is the [issue](https://github.com/Expensify/App/issues/2151), you can check the description [here](https://github.com/Expensify/App/issues/2151#issuecomment-810630944)
2. at that time we were not using the KeyboardAvoidingView for the ScreenWrapper, please check this file https://github.com/aliabbasmalik8/Expensify.cash/edit/IS-2151-fab-hidden-issue/src/components/ScreenWrapper.js
3. now we are using  KeyboardAvoidingView under ScreenWrapper, so no need to wrap the FAB component under KeyboardAvoidingView, as we are wrapping `SidebarScreen` under `ScreenWrapper`, and its using `KeyboardAvoidingView`



> Also your proposal seems like redundant of Solution 1 [#10905 (comment)](https://github.com/Expensify/App/issues/10905#issuecomment-1248953829)
> 
I missed checking [here](https://github.com/Expensify/App/issues/10905#issuecomment-1248953829) but I think that this is the solution 

> 
> video that shows that after removing file, its working fine in iPhone 13 and iPad pro
> 
> https://drive.google.com/file/d/12HCy-tr3GH4Qn-oWR_cZLi2UEcJlnJ9Q/view?usp=sharing

@Santhosh-Sellavel can you please check this video and give feedback if this video is satisfied the requirement or not, as I think this is as per this [comment](https://github.com/Expensify/App/issues/10905#issuecomment-1246997790)

**Comment by Santhosh-Sellavel** [comment]
Thanks for the proposals.

@jayeshmangwani 
> I missed checking https://github.com/Expensify/App/issues/10905#issuecomment-1248953829 but I think that this is the solution

If that is the case we should go with the first one if that proposal is accepted.

But before proceeding, @tgolen this issue also revolves around the keyboard, can you verify this too got fixed? Can you confirm, please?
Thanks!

**Comment by jliexpensify** [comment]
Bump @tgolen to answer @Santhosh-Sellavel 's question!

**Comment by tgolen** [comment]
I'm going to place a HOLD on this and add the issue to this master tracking issue: https://github.com/Expensify/App/issues/10273

I think some of the problems should go away once that is done.

**Comment by ctkochan22** [comment]
On hold

**Comment by tgolen** [comment]
I removed the HOLD on this and assigned it to me because this is fixed in https://github.com/Expensify/App/pull/11586

**Comment by jliexpensify** [comment]
@tgolen - thanks! Since it's fixed by another PR, could we remove this Upworks job?

**Comment by tgolen** [comment]
Yeah, it's fine to remove it.

**Comment by jliexpensify** [comment]
Upworks job removed

**Comment by tgolen** [comment]
PR is still in progress

**Comment by jliexpensify** [comment]
@tgolen - this is now an Internal issue right? Could I remove myself and @Santhosh-Sellavel from this GH?

**Comment by tgolen** [comment]
That proposal sounds great! 🟢 from me

**Comment by jliexpensify** [comment]
Sorry @tgolen I'm a little confused: did you need a $750 Upworks job created for https://github.com/Expensify/App/pull/11586? And Santhosh and Aimane invited?

I'm just trying to work out if I need to do anything...thanks.

**Comment by tgolen** [comment]
@jliexpensify let's move that convo over to #11586.

**Comment by tgolen** [comment]
As far as this issue is concerned, I think it basically totally reset when I took it off hold. We need a $1000 job created in UpWork and hire @aimane-chnaif for the proposal they gave right above this.

**Comment by Santhosh-Sellavel** [comment]
@tgolen 
Just reviewed it works well on iPad & Tab. Workswell Android Tab Web also But iPad Safari alone seems to be sloppy as it was earlier. 

https://user-images.githubusercontent.com/85645967/199301952-4b2dfca2-2d2c-4840-b243-15f451d8a81d.mp4

**Comment by Santhosh-Sellavel** [comment]
Seem the PR is already up for this one if we are good with the above let's assign this to @aimane-chnaif, thanks!

**Comment by jliexpensify** [comment]
> We need a $1000 job created in UpWork and hire @aimane-chnaif for the proposal they gave right above this.

Thanks for clarifying @tgolen - I'll get this sorted now.

**Comment by jliexpensify** [comment]
Internal - https://www.upwork.com/ab/applicants/1587570510876553216/job-details
External - https://www.upwork.com/jobs/~0137afe133f0fca3ca

Hired @aimane-chnaif and @Santhosh-Sellavel

**Comment by tgolen** [comment]
@Santhosh-Sellavel Thanks for reporting that! I think it's OK for this issue. I noticed there is another issue for handling view rotation on iPad better, and that's probably more related to the bug you're seeing in iPad safari: https://github.com/Expensify/Expensify/issues/136195

**Comment by JmillsExpensify** [comment]
@jliexpensify Given that we're still early and no proposals have been accepted, I'm assigning myself per the new BZ process: https://stackoverflow.com/c/expensify/questions/14597/14598#14598.

**Comment by jliexpensify** [comment]
@JmillsExpensify - Happy for you to assign this one over to yourself, however, my understanding was that we had an [approved proposal](https://github.com/Expensify/App/issues/10905#issuecomment-1296382891) from @aimane-chnaif

**Comment by JmillsExpensify** [comment]
The Upwork job was just created so I figured no biggie, but please take it back if you'd like!

**Comment by JmillsExpensify** [comment]
@jliexpensify Don't forget to get the BZ checklist kicked off above please!

**Comment by JmillsExpensify** [comment]
@jliexpensify Just wanted to circle back and make sure you've gone through the linked BZ checklist?

**Comment by jliexpensify** [comment]
@JmillsExpensify [I previously posted here around what parts of the checklist I should be doing](https://expensify.slack.com/archives/C01SKUP7QR0/p1667775111039369), and you mentioned changes were coming. 

I'm still confused here: am I doing the entire checklist? The first 3 items sound like they should be reviewed by the primary engineer (Tim), whilst the last few items are related to payment, which isn't due yet.

**Comment by JmillsExpensify** [comment]
@jliexpensify I don't recall saying don't fill out the checklist, as any changes to Melvin are unfortunately not retro-active. All good, here's an [example checklist](https://github.com/Expensify/App/issues/10824#issuecomment-1310855916) I got on Friday. That should clear up your questions.

**Comment by jliexpensify** [comment]
Thanks @JmillsExpensify, appreciate you sharing that checklist. I've updated the one above to mirror yours.

Regression test question posted here - https://expensify.slack.com/archives/C01SKUP7QR0/p1668379720583589

**Comment by jliexpensify** [comment]
@Santhosh-Sellavel and @tgolen - I've done the regression test GH, and have tagged you both in the checklist. Cheers!

**Comment by tgolen** [comment]
I've filled out the first two checkboxes for identifying the PR. @Santhosh-Sellavel would you like to create a slack thread for discussing if any changes are necessary for the checklist?

**Comment by JmillsExpensify** [comment]
Quick follow-up question: @jliexpensify has payment been issued to contributors?

**Comment by Santhosh-Sellavel** [comment]
Thread started [here](https://expensify.slack.com/archives/C049HHMV9SM/p1669053920014679)

**Comment by jliexpensify** [comment]
Sorry @JmillsExpensify - dropped the ball on this one! I'll sort out payment now.

**Comment by jliexpensify** [comment]
@Santhosh-Sellavel , @aimane-chnaif and @thesahindia apologies for the delay: once you accept, I'll pay out $1000 to each of you.

**Comment by thesahindia** [comment]
Accepted, thanks!

>  I'll pay out $1000 to each of you.

I believe it's $250 for me (reporting compensation)

**Comment by jliexpensify** [comment]
@thesahindia - thanks for pointing that out and your honesty! My new dad brain appreciates you.

Everyone is paid, feedback has been left and the job closed in Upworks. I'll keep this GH open for @Santhosh-Sellavel to update on the discussion. Thanks!

**Comment by JmillsExpensify** [comment]
Thank you @jliexpensify! Echoing your last comment, this is the last todo before we can close this issue.

> @Santhosh-Sellavel @tgolen A discussion in #contributor-plus has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

**Comment by tgolen** [comment]
@jliexpensify the BugZero checklist is finished now so this issue is OK to close out now. I'll let you do the honors.

**Comment by jliexpensify** [comment]
Nice work, thanks @tgolen , @aimane-chnaif , @Santhosh-Sellavel, @shawnborton and @JmillsExpensify !
